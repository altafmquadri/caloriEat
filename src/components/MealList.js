import React from 'react';
import { connect } from 'react-redux'
import MealItem from './MealItem'

const MealList = (props) => {
    return (
        <div>
            {!props.meals.length ? (
                <div>
                    <span>No Meals</span>
                </div>
            ) : (
                    props.meals.map(meal =>
                        <MealItem key={meal.uuid} {...meal} />
                    )
                )}
        </div>
    );
}

const mapStateToProps = state => ({
    meals: state.meals
})

export default connect(mapStateToProps)(MealList);
