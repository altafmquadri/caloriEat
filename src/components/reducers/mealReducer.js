const mealReducer = (state = [], action) => {
    switch (action.type) {
        case 'ADD_MEAL': {
            const meal = action.payload
            return [...state, meal]
        }
        case 'EDIT_MEAL': {
            const {
                calories,
                totalNutrients: {
                    CHOCDF: { quantity: carbohydrates },
                    FAT: { quantity: fat },
                    FIBTG: { quantity: fiber },
                    PROCNT: { quantity: protein },
                }
            } = action.payload
            const item = {
                ...state.find(meal => meal.uuid === action.uuid),
                ...action.properties,
                calories,
                carbohydrates: parseFloat(carbohydrates).toFixed(2),
                fat: parseFloat(fat).toFixed(2),
                fiber: parseFloat(fiber).toFixed(2),
                protein: parseFloat(protein).toFixed(2),
            }
            return state.map(meal => {
                if (meal.uuid === action.uuid) {
                    return meal = { ...meal, ...item }
                }
                return meal
            })
        }
        case 'DELETE_MEAL': {
            const id = action.payload
            return state.filter(meal => meal.uuid !== id)
        }
        default:
            return state
    }
}

export default mealReducer