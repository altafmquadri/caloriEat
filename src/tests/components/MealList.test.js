import React from 'react';
import { shallow } from 'enzyme'
import MealList from '../../components/MealList'
import { meals } from '../fixtures/mealReducer'

describe('<MealList/>', () => {
    test('renders MealList without any meals', () => {
        const wrapper = shallow(<MealList meals={[]} />)
        expect(wrapper.find('span').text()).toBe('No Meals')
        expect(wrapper).toMatchSnapshot()
    })
    test('renders MealList with meals', () => {
        const wrapper = shallow(<MealList meals={meals} />)
        expect(wrapper).toMatchSnapshot()
        
    })
})