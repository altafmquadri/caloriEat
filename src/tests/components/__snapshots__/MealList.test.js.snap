// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealList/> renders MealList with meals 1`] = `
<div>
  <MealItem
    calories={148}
    carbohydrates="11.71"
    date="2020-08-17T18:36:13.173Z"
    fat="7.93"
    fiber="0.00"
    foodId="food_b49rs1kaw0jktabzkg2vvanvvsis"
    foodItem="milk"
    image="https://www.edamam.com/food-img/7c9/7c9962acf83654a8d98ea6a2ade93735.jpg"
    key="1"
    mealCategory="breakfast"
    measureURI="http://www.edamam.com/ontologies/edamam.owl#Measure_cup"
    measures={
      Array [
        Object {
          "label": "Serving",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_serving",
        },
        Object {
          "label": "Gram",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_gram",
        },
        Object {
          "label": "Ounce",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_ounce",
        },
        Object {
          "label": "Pound",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_pound",
        },
        Object {
          "label": "Kilogram",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_kilogram",
        },
        Object {
          "label": "Tablespoon",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_tablespoon",
        },
        Object {
          "label": "Cup",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_cup",
        },
        Object {
          "label": "Bottle",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_bottle",
        },
        Object {
          "label": "Fluid ounce",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_fluid_ounce",
        },
        Object {
          "label": "Quart",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_quart",
        },
        Object {
          "label": "Gallon",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_gallon",
        },
      ]
    }
    protein="7.69"
    quantity={1}
    uuid="1"
  />
  <MealItem
    calories={55}
    carbohydrates="8.52"
    date="2020-08-17T18:36:36.867Z"
    fat="2.30"
    fiber="0.35"
    foodId="food_bihs8eobc58jvqbhmt6s3bym1rhf"
    foodItem="oreo"
    image="https://www.edamam.com/food-img/a90/a9011c79689c79db4f1a64776509dc69.jpg"
    key="2"
    mealCategory="snack"
    measureURI="http://www.edamam.com/ontologies/edamam.owl#Measure_cookie"
    measures={
      Array [
        Object {
          "label": "Whole",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_unit",
        },
        Object {
          "label": "Serving",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_serving",
        },
        Object {
          "label": "Cookie",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_cookie",
        },
        Object {
          "label": "Oreo",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_oreo",
        },
        Object {
          "label": "Sandwich",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_sandwich",
        },
        Object {
          "label": "Box",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_box",
        },
        Object {
          "label": "Container",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_container",
        },
        Object {
          "label": "Package",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_package",
        },
        Object {
          "label": "Gram",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_gram",
        },
        Object {
          "label": "Ounce",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_ounce",
        },
        Object {
          "label": "Pound",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_pound",
        },
        Object {
          "label": "Kilogram",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_kilogram",
        },
      ]
    }
    protein="0.63"
    quantity={1}
    uuid="2"
  />
  <MealItem
    calories={255}
    carbohydrates="40.76"
    date="2020-08-17T18:37:00.048Z"
    fat="8.49"
    fiber="3.40"
    foodId="food_aqa4qtwapci86kar0l275bjqmkal"
    foodItem="french fries"
    image="https://www.edamam.com/food-img/559/559b143365bf3dde8d5889ec627e1da7.jpg"
    key="3"
    mealCategory="lunch"
    measureURI="http://www.edamam.com/ontologies/edamam.owl#Measure_ounce"
    measures={
      Array [
        Object {
          "label": "Serving",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_serving",
        },
        Object {
          "label": "Strip",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_strip",
        },
        Object {
          "label": "Gram",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_gram",
        },
        Object {
          "label": "Ounce",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_ounce",
        },
        Object {
          "label": "Pound",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_pound",
        },
        Object {
          "label": "Kilogram",
          "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_kilogram",
        },
      ]
    }
    protein="3.98"
    quantity={6}
    uuid="3"
  />
</div>
`;

exports[`<MealList/> renders MealList without any meals 1`] = `
<div>
  <div>
    <span>
      No Meals
    </span>
  </div>
</div>
`;
